### DISCUSSION TITLE
Does anyone know how to properly set Content Security Policy for Web3Modal?

### DISCUSSION THREAD

**Discussion by hskang9** [WalletConnect/walletconnect-monorepo#2084] [Developer Support]
I need to figure out which image src they use to render networks so that I can add it in `img-src` CSP.

**Comment by hskang9** [comment]
Is Walletconnect team allow hackers XSS attack on the website with turning on 'unsafe-inline' and 'unsafe-eval'? Seriously?

**Reply by xzilja** [reply]
- Images are pulled from Staging env / api to test “registry_submissions” in web3modal, needs to be scoped to project_id do not overexpose data.
- What do you need to use `unsafe-eval` for?
- `unsafe-inline` will be needed by most ui libs to inject custom styles
- re xss attack concern, csp rules apply only to libraries used in your app, so add [subresource integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) check to ensure you are using trusted sources. Every other xss attack vector is up to you to secure against.

**Reply by xzilja** [reply]
Also, for any other network related questions, you can always use browser inspector to see what's being fetched
